services/wfm/src/main/java/org/openkilda/wfm/share/hubandspoke/CoordinatedBolt.java

```diff
@@ -40,22 +41,19 @@
     }
 
     @Override
-    public void execute(Tuple input) {
-        log.debug("{} input tuple from {}:{} size {}",
-                getClass().getName(), input.getSourceComponent(), input.getSourceStreamId(), input.size());
-        try {
-            if (CoordinatorBolt.ID.equals(input.getSourceComponent())) {
-                String key = input.getStringByField(MessageTranslator.KEY_FIELD);
-                onTimeout(key);
-            } else {
-                handleInput(input);
-            }
-        } catch (Exception e) {
-            log.error(String.format("Unhandled exception in %s", getClass().getName()), e);
-        } finally {
-            if (autoAck) {
-                ack(input);
-            }
+    public void dispatch(Tuple input) throws AbstractException {
+        if (CoordinatorBolt.ID.equals(input.getSourceComponent())) {
+            String key = input.getStringByField(MessageTranslator.KEY_FIELD);
+            onTimeout(key);
+        } else {
+            super.dispatch(input);
+        }
+    }
+
+    @Override
+    public void ack(Tuple input) {
```

Comment:
tbh I don't like suck ack implementation of public method that will do something if and only if I set flag to true. This is error leading way as for me. Maybe it's worth to improve this thing.